src/utils/miscellaneous.ts

```diff
@@ -12,9 +12,9 @@ export function getToken<T extends TokenDetails, K extends keyof T>(
 ): T | undefined {
   const valueUppercase = value.toUpperCase()
   return (tokens || []).find((token: T): boolean => {
-    const value = token[key]
-    if (value) {
-      return value.toString().toUpperCase() === valueUppercase
+    const tokenKeyValue = token[key]
+    if (tokenKeyValue) {
+      return typeof tokenKeyValue === 'string' && tokenKeyValue.toString().toUpperCase() === valueUppercase
```

Comment:
Can you do the type guard in the if?
In that case I think the `toString()` is not needed, right?

```typescript
    if (tokenKeyValue && typeof tokenKeyValue === 'string') {
￼      return tokenKeyValue.toUpperCase() === valueUppercase
```